packages/client/src/views/VerifyCertificate/VerifyCertificatePage.tsx

```diff
@@ -246,6 +247,21 @@ export function VerifyCertificatePage() {
     if (isDeathRegistration(data)) return data.deceased?.gender
   }
 
+  const getLastCertifiedDate = (data: RecordDetails) => {
+    // find when declaration was certified
+    return (
+      data.history &&
+      data.history.find(
+        (h, index, allHistory) =>
+          index - 1 &&
+          allHistory[index]?.regStatus?.toString() ===
+            SUBMISSION_STATUS.CERTIFIED &&
+          allHistory[index + 1]?.regStatus?.toString() !==
+            SUBMISSION_STATUS.CERTIFIED
+      )
+    )?.date
+  }
```

Comment:
This assumes the tasks are in ascending order according to lastModified time. Then shouldn't just looking for the first "CERTIFIED" item suffice?

Reply:
Absolutely! No need to check the status change